src/parseTools.js

```diff
@@ -1354,8 +1354,11 @@ function convertPtrToIdx(ptr, accessWidth) {
 
   let conversion;
   if (MEMORY64) {
-    if (MAXIMUM_MEMORY > MAX_MEMORY53) conversion = /^\d+$/.test(accessWidth) ? `${ptr} >>= ${accessWidth}n` : `${ptr} >>= BigInt(${accessWidth})`;
-    else conversion = `${ptr} = Number(${ptr}) / ${accessWidth}`;
+    if (MAXIMUM_MEMORY > MAX_MEMORY53) {
+      conversion = /^\d+$/.test(accessWidth) ? `${ptr} >>= ${accessWidth}n` : `${ptr} >>= BigInt(${accessWidth})`;
+    } else {
+      conversion = `${ptr} = Number(${ptr}) / ${accessWidth}`;
```

Comment:
i am currently looking into getting a wasm64 build to work with the emscripten opengl integration and have stumbled over this pull request. While the approach looks good, there is an issue with this line. I assume that accessWidth is supposed to be the number of bits to shift. So you should not divide ptr by this but instead divide it by 2^accessWidth.

Reply:
Good catch, that is definitely wrong. I have been looking to get back to finishing this PR soon, once I get my other audio worklets feature through the finish line. I'll push a fix to this in a sec.

Reply:
Thanks for the quick fix!